src/modules/roc_sdp/parser.rl

```diff
@@ -0,0 +1,310 @@
+
+            
+            # In session-level: c=<nettype> <addrtype> <connection-address>
+            session_connection_nettype = "IN";
+            
+            # Either IPv6 or IPv4/TTL
+            session_connection_address = non_ws_string >start_token
+                %set_session_connection_data;
+
+            session_connection_with_addrtype =  
+                ( "IP4" %{ cur_addrtype = address::Family_IPv4; } 
+                    SP session_connection_address '/' digit+
+                | "IP6" %{ cur_addrtype = address::Family_IPv6; } 
+                    SP session_connection_address
+                );
+
+            session_connection_data = session_connection_nettype SP 
+                session_connection_with_addrtype;
+        
+            # Each media description starts with an "m=" field and is terminated by
+            # either the next "m=" field or by the end of the session description
+            # m=<type> <port> <proto> <fmt>
+
+            # Typically "audio", "video", "text", or "application"
+            media_type = ("audio"  %{ result.last_media_description()->set_type(sdp::MediaType_Audio); } 
+                | "video" %{ result.last_media_description()->set_type(sdp::MediaType_Video); }
+                | "text" %{ result.last_media_description()->set_type(sdp::MediaType_Text); }
+                | "application" %{ result.last_media_description()->set_type(sdp::MediaType_Application); }
+            );
+                
+            
+            # Typically "RTP/AVP".
+            media_proto = "RTP/AVP"  %{ result.last_media_description()->set_proto(sdp::MediaTransport_RTP_AVP); };
+
+            media_port = digit+ >start_token %set_media_port;
+            
+            media_nb_ports = digit+ >start_token %set_media_port;
+
+            # Typically an RTP payload id for audio and video media
+            media_fmt = token >start_token %add_media_payload_id;
+
+            media_description = (media_type SP media_port ('/' media_nb_ports)? SP media_proto (SP media_fmt)+)
+                >add_media;
+            
+            media_field = 'm='i media_description;
+
+            # In media-level: c=<nettype> <addrtype> <connection-address>
+            media_connection_nettype = "IN";
+            
+            # Either IPv6 or IPv4/TTL
+            media_connection_address = non_ws_string >start_token 
+                %add_media_connection_data;
+
+            media_connection_with_addrtype =  
+                ( "IP4" %{ cur_addrtype = address::Family_IPv4; } 
+                    SP media_connection_address '/' digit+
+                | "IP6" %{ cur_addrtype = address::Family_IPv6; } 
+                    SP media_connection_address
+                );
+
+            media_connection_data = media_connection_nettype SP 
+                media_connection_with_addrtype;
+
+            media_connection_field = 'c='i media_connection_data;
```

Comment:
Could we avoid duplicating grammar for media- and session-level connection data?